src/Serialization/containers.jl

```diff
@@ -53,9 +46,9 @@ function save_type_params(s::SerializerState,
       for (k, v) in params(tp)
         if k == :dims
           save_object(s, v, k)
-          continue
+        else
+          save_type_params(s, v, k)
```

Comment:
Looking at this code, I wonder: why is there `save_type_params`? Could most of those be turned into `save_object` methods (and similar for `load`)?

Just these two would be dropped without direct replacement:
```
function save_type_params(s::SerializerState, obj::Any, key::Symbol)
  set_key(s, key)
  save_type_params(s, obj)
end

function save_type_params(s::SerializerState, obj::T) where T
  save_type_params(s, type_params(obj))
end
```
and then some `save_type_params(s, obj)` will have to be replaced by `save_object(s, type_params(obj))`